GVFS/GVFS.Common/NuGetUpgrade/ModernNuGetUpgrader.cs

```diff
@@ -0,0 +1,256 @@
+using GVFS.Common;
+using GVFS.Common.FileSystem;
+using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using System;
+using System.Net.Http;
+using System.Runtime.InteropServices;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+
+namespace GVFS.Common.NuGetUpgrade
+{
+    public class ModernNuGetUpgrader : NuGetUpgrader
+    {
+        private HttpClient httpClient;
+        private string platform;
+
+        public ModernNuGetUpgrader(
+           string currentVersion,
+           ITracer tracer,
+           PhysicalFileSystem fileSystem,
+           HttpClient httpClient,
+           bool dryRun,
+           bool noVerify,
+           ModernNuGetUpgraderConfig config,
+           string downloadFolder,
+           string platform,
+           ICredentialStore credentialStore)

```

Comment:
The number of parameters for this class is a little out of control.  Can we have something like a `NuGetUpgraderOptions` that has the currentVersion, dryRun, noVerify, downloadFolder, platform, and maybe the values from the config to trim this down?

Reply:
I agree - I have some thoughts on how to tame this, but I would like to tackle that in a seperate PR.